### DISCUSSION TITLE
Question about label propagation

### DISCUSSION THREAD

**Discussion by xnuohz** [pyg-team/pytorch_geometric#2565] [Q&A]
In your implementation of [label propagation](https://github.com/rusty1s/pytorch_geometric/blob/e6b8d6427ad930c6117298006d7eebea0a37ceac/torch_geometric/nn/models/label_prop.py#L54), do we need to update `res` for each propagation? Since `res` is always the initial `y` in each propagation.

**Comment by rusty1s** [comment]
I guess this depends on how you implement the Label Propagation model. We follow [this code](https://github.com/CUAI/CorrectAndSmooth/blob/master/outcome_correlation.py#L160-L173) to implement it.

In general, adding residuals to the initial labels makes a lot of sense, as it avoids that labels are smoothed too heavily, e.g., in each iteration we re-inject the ground-truth labels to the output to some extend. This will help the model to make better predictions on nearby nodes in homophily graphs.

**Reply by xnuohz** [reply]
ok, thanks:)
